src/math/geometry.cpp

```diff
@@ -161,7 +161,7 @@ ConvexityResult isConvexQuad(std::array<Eigen::VectorXd, 4> coords)
   Eigen::Vector3d normalVector = e_1.cross(e_2);
 
   PRECICE_CHECK(math::equals(normalVector.dot(coords[3] - coordOrigin), 0.0),
-                "Non-planar quads are not supported. The vertex coordinates are: {}.", coords);
+                "Non-planar quads are not supported. The vertex coordinates are: {}; {}; {}; {}.", coords[0], coords[1], coords[2], coords[3]);
```

Comment:
This should still work due to:

https://github.com/precice/precice/blob/0590b349df3e28e0a9a99e4e5e1a05256565d494/src/utils/fmtSTL.hpp#L28-L36

Reply:
There is apparently a `const` issue with the logger (which is defined in the source file?!). I don't get it fully:
```cpp
error: passing ‘const formatter_type’ {aka ‘const fmt::v9::formatter<Eigen::Matrix<double, -1, 1>, char, void>’} as ‘this’ argument discards qualifiers [-fpermissive]
```

Reply:
Sounds like `format()` needs to be `const` to allow composition?

Could you try?

```diff
-   auto format(const std::array<T, n> &v, FormatContext &ctx) 
+   auto format(const std::array<T, n> &v, FormatContext &ctx) const
```

Reply:
Ah now I got it. Yes, works! Do we need to update the other fmtSTL formatter to be `const` as well? Either we are not using them or we don't need to update, I guess it is the first one.

Reply:
Doesn't work either.

Reply:
I think the real issue here is that we have a nested sequence of array-like structures, namely a Vector of arrays. Using a simple array works with the current implementation.

Reply:
I found a solution and pushed it.
The entire chain needs to be const.